.github/workflows/task-test-rust.yml

```diff
@@ -0,0 +1,45 @@
+---
+name: Task - Test Rust
+
+on:
+  workflow_dispatch:
+  workflow_call:
+
+jobs:
+  test-rust:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      # Setup the environment with all necessary dependencies
+      - uses: ./.github/actions/load-env
+
+      - uses: ./.github/actions/rust-setup
+        with:
+          cache-key: ${{ env.BUILD_RUST_CACHE_KEY }}
+          rust-version: ${{ env.BUILD_RUST_VERSION }}
+          scarb-version: ${{ env.BUILD_SCARB_VERSION }}
+          install-mold: true
+          install-scarb: true
+          install-foundry: true
+          foundry-version: ${{ env.BUILD_FOUNDRY_VERSION }}
+
+      - name: Install cargo-nextest
+        uses: taiki-e/install-action@nextest
+
+      - name: Cache nextest
+        uses: actions/cache@v4
+        with:
+          path: |
+            target/nextest
+            target/nextest-*
+          key: ${{ runner.os }}-rust-test-${{ hashFiles('madara/**/*.rs','**/Cargo.lock') }}
+          restore-keys: |
+            ${{ runner.os }}-rust-test-${{ hashFiles('**/Cargo.lock') }}
+            ${{ runner.os }}-rust-test-
+
+      - name: Run unit tests
+        run: |
+          cargo nextest run --release --workspace --test-threads=1
```

Comment:
Would be nice to have in the .env file the variable for the number of threads to use in nextest, just to able to bump it if we need to.